src/tests/common/mod.rs

```diff
@@ -0,0 +1,162 @@
+pub mod constants;
+
+use std::str::FromStr;
+
+use axum::async_trait;
+use diesel::{Connection, PgConnection, RunQueryDsl};
+use diesel_async::pooled_connection::deadpool::Pool;
+use diesel_async::pooled_connection::AsyncDieselConnectionManager;
+use diesel_async::{AsyncPgConnection, RunQueryDsl as AsyncRunQueryDsl};
+use uuid::Uuid;
+
+use crate::domain::models::indexer::{IndexerModel, IndexerStatus, IndexerType};
+use crate::infra::errors::InfraError;
+use crate::infra::repositories::indexer_repository::{
+    IndexerFilter, NewIndexerDb, Repository, UpdateIndexerStatusAndProcessIdDb, UpdateIndexerStatusDb,
+};
+use crate::run_migrations;
+
+// Keep the database info in mind to drop them later
+pub struct TestContext {
+    pub base_url: String,
+    pub db_name: String,
+    pub pool: Pool<AsyncPgConnection>,
+}
+
+impl TestContext {
```

Comment:
Why do we need a `TestContext`, can we just create a dummy pool to initialize the repository? Also, how will this work in like Github actions?